Update CSES Flight Discount.mdx (#251)
cpinitiative/usaco-guide: solutions/CSES Flight Discount.mdx

title: CSES - Flight Discount
author: Kai Wang
---

> This is a meet in the middle problem. Take the obvious graph interpretation.

Author: Good point and I don't know... Here is supporting evidence from geeksforgeeks.com: "Like divide and conquer it splits the problem into two, solves them individually and then merge them"

> This is a meet in the middle problem. Take the obvious graph interpretation.

Author: Maybe delete "take the obvious graph interpretation..." it's too formal and unnecessary.

> This is a meet in the middle problem. Take the obvious graph interpretation.

Reviewer: I think both sentences are deleted...

> Say I use the discount on edge between A and B, and I use the edge.

Author: Yes that is clearer.

> There are two cases: I have to go from $1\rightarrow A\rightarrow B\rightarrow N$, or $1\rightarrow B\rightarrow A\rightarrow N$. It turns out that we need the distance between 1 and A, and N and B.

Author: yes

> This means we use dijkstra to compute the distance from 1, N to every vertex, then compute $\text{min}_{A\rightarrow B} dist1[A]+c(A,B)+distN[B]$,

Reviewer: ```suggestion
We can use Dijkstra's to compute the distance from $1$ and $N$ to every vertex. Then our answer is $\text{min}_{A\rightarrow B} dist1[A]+c(A,B)+distN[B]$,
```

Author: Yes

> where $c(A,B)$ is the cost to travel from city A to city B.

Reviewer: ```suggestion
where $c(A,B)$ is the cost to travel from city A to city B after applying the coupon to that flight.
```

Author: Oh yes that was a typo oops
